fullscreen.bs

```diff
@@ -68,13 +71,14 @@ is an <a>ordered set</a> of (<a>string</a>, <a>element</a>) <a>tuples</a>. It is
 
  <li><p>Set <var>element</var>'s <a>fullscreen flag</a>.
 
- <li><p><a for="top layer">Add</a> <var>element</var> to <var>element</var>'s <a>node document</a>'s
- <a>top layer</a>.
+ <li><p><a>Request removal from the top layer</a> given <var>element</var>.
+
+ <li><a>Add to the top layer</a> given <var>element</var>.
```

Comment:
Only the second step is needed here looking at the algorithm for "add".

Reply:
Without the call to "request removal from the top layer", the assert would be hit, right?

Reply:
I see. So we preserve the behavior but allow for slightly novel behavior in new APIs. I guess that's okay. If we don't end up wanting new behavior though and I think @nt1m argued we didn't for perf reasons we should simplify this again.

Perhaps it's best to wait for @nt1m and merge this tomorrow. I can ping him.

Reply:
I'm not sure what you mean by "allow for slightly novel behavior in new APIs". All APIs would cycle thru the algorithms in this same way, unless they wanted to get into the guts and directly manipulate the top layer list (which they shouldn't).

The only two behaviors that specs should end up landing on for "put something in the top layer when it's already there" are this (cycle it to the end of the list) and "fail", probably throwing an error when they do so.

Reply:
It does kinda seem like unless they would always remove prior to adding they'd have to perform a contains check. Which would be the somewhat novel behavior, but not entirely novel as the old API technically allowed for that as well.

Reply:
Right, specs have to either check if the element is already in top layer and treat it as an error, or call "request removal from the top layer" to get the moving behavior. The latter preserves the behavior mentioned in the last note in https://fullscreen.spec.whatwg.org/commit-snapshots/afd56a35f409e5595dd861f41390a1016ebd6aa2/#new-stacking-layer.